### PR TITLE
Respect env.ALL_CHANGED_FILES if present

### PR DESCRIPTION
**Pull request by jaredly**
## Summary:
If it's absent, the fall back to the method that might give false positives.

Issue: https://khanacademy.atlassian.net/browse/FEI-3796

## Test plan:
This pull-request shows the new code in action https://github.com/Khan/eslint-action/pull/27/files

### CODE REVIEW

File: git-changed-files.js [hunk 1]
```diff
@@ -62,12 +62,36 @@ const isFileIgnored = (workingDirectory, file) => {
...
+    // changed files list. Otherwise, we fallback to a simple diff between the
+    // current and base branch, which might give false positives if the base
+    // is ahead of the current branch.
+    //
+    //   - name: Get All Changed Files
+    //     uses: jaredly/get-changed-files@absolute
+    //     id: changed
+    //     with:
+    //       format: 'json'
+    //       absolute: true
```
Comment (jeresig) [inline git-changed-files.js:84]: I assume this is coming in another diff - but why the change to absolute file paths and outputting as JSON?

File: git-changed-files.js [hunk 2]
```diff
@@ -62,12 +62,36 @@ const isFileIgnored = (workingDirectory, file) => {
...
+    //
+    //   - uses: allenevans/set-env@v2.0.0
+    //     with:
+    //       ALL_CHANGED_FILES: '${{ steps.changed.outputs.added_modified }}'
+    //
+    if (process.env.ALL_CHANGED_FILES) {
+        const files = JSON.parse(process.env.ALL_CHANGED_FILES)
+        return files.filter(path => !isFileIgnored(cwd, path))
```
Comment (jeresig) [inline git-changed-files.js:92]: You may have to manually run the build/lint process (I see that Prettier hasn't run here).

**Review by jeresig — APPROVED** [review]

File: git-changed-files.js (hunk 2 above)
Comment (jaredly) [reply git-changed-files.js:92]: @jeresig it is running, we just don't have it configured on this repo, so it's using defaults that don't add semicolons 🙃
I'll add a prettierrc

Comment (jeresig) [reply git-changed-files.js:92]: Oh whoops! haha

File: git-changed-files.js (hunk 1 above)
Comment (jaredly) [reply git-changed-files.js:84]: So:
- outputting as json is more robust
- the get-changed-files action produces files relative to the github base by default; I [added the absolute option](https://github.com/jitterbit/get-changed-files/pull/40) because that's what the rest of our system expects, and it allows us to run commands from various subdirectories, not just the workspace root.
